File: tools/verify_supabase.js
import 'dotenv/config';
import { createClient } from '@supabase/supabase-js';

const supabaseUrl = process.env.VITE_SUPABASE_URL || '';
const supabaseKey = process.env.VITE_SUPABASE_ANON_KEY || '';

if (!supabaseUrl || !supabaseKey) {
    console.log("❌ Missing VITE_SUPABASE_URL or VITE_SUPABASE_ANON_KEY");
    process.exit(1);
}

const supabase = createClient(supabaseUrl, supabaseKey);

async function verify() {
    console.log("Connecting to:", supabaseUrl);
    try {
        const { data, error } = await supabase.from('plots').select('*').limit(1);

        if (error) {
            console.log("❌ Supabase connection error:", error.message);
            process.exit(1);
        }

        console.log("✅ Successfully connected to Supabase 'plots' table.");
        console.log("Data:", data);
    } catch (err) {
        console.log("❌ Connection failed.", err);
    }
}

verify();

File: tools/verify_supabase.py
import os
from supabase import create_client, Client
from dotenv import load_dotenv

load_dotenv()

url: str = os.environ.get("VITE_SUPABASE_URL")
key: str = os.environ.get("VITE_SUPABASE_ANON_KEY")

if not url or not key:
    print("❌ Supabase URL or Key not found in .env")
    exit(1)

supabase: Client = create_client(url, key)

try:
    response = supabase.table('plots').select("*").limit(1).execute()
    print("✅ Successfully connected to Supabase 'plots' table.")
    print(f"Current data: {response.data}")
except Exception as e:
    print(f"❌ Failed to connect to Supabase: {e}")
